androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToPredictions.kt

```diff
@@ -3,29 +3,31 @@ package com.mbta.tid.mbta_app.android.state
 import android.util.Log
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.collectAsState
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.setValue
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
-import androidx.lifecycle.asFlow
 import androidx.lifecycle.compose.LifecycleResumeEffect
 import androidx.lifecycle.viewmodel.compose.viewModel
 import com.mbta.tid.mbta_app.model.response.ApiResult
 import com.mbta.tid.mbta_app.model.response.PredictionsByStopJoinResponse
 import com.mbta.tid.mbta_app.model.response.PredictionsByStopMessageResponse
 import com.mbta.tid.mbta_app.model.response.PredictionsStreamDataResponse
 import com.mbta.tid.mbta_app.repositories.IPredictionsRepository
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.debounce
 import kotlinx.coroutines.flow.map
 import org.koin.compose.koinInject
 
 class PredictionsViewModel(
     private val predictionsRepository: IPredictionsRepository,
 ) : ViewModel() {
-    private val _predictions: MutableLiveData<PredictionsByStopJoinResponse> = MutableLiveData()
-    val predictions: LiveData<PredictionsByStopJoinResponse> = _predictions
-    val predictionsFlow = predictions.asFlow().map { it.toPredictionsStreamDataResponse() }
+    private val _predictions = MutableStateFlow<PredictionsByStopJoinResponse?>(null)
+    private val _counter = MutableStateFlow<Int>(0)
+
+    val predictions: StateFlow<PredictionsByStopJoinResponse?> = _predictions
+    val predictionsFlow =
+        predictions.map { it?.toPredictionsStreamDataResponse() }.debounce(0.1.seconds)
```

Comment:
Is the ordering of `map` vs. `debounce` relevant?

Reply:
Oh good point, I'll swap to the two to avoid excessive maps